source/ice_api.c

```diff
@@ -0,0 +1,1676 @@
+
+static IceResult_t Ice_CheckPeerReflexiveCandidate( IceAgent_t * pIceAgent,
+                                                    IceIPAddress_t * pIpAddr,
+                                                    IceCandidate_t * pLocalCandidate,
+                                                    uint32_t priority,
+                                                    bool isRemote )
+{
+    IceResult_t retStatus = ICE_RESULT_OK;
+    IceCandidate_t * pPeerReflexiveCandidate;
+    bool foundDuplicatedCandidate = false;
+
+    if( ( pIceAgent == NULL ) ||
+        ( pIpAddr == NULL ) ||
+        ( !isRemote && ( pLocalCandidate == NULL ) ) )
+    {
+        retStatus = ICE_RESULT_BAD_PARAM;
+    }
+
+    if( retStatus == ICE_RESULT_OK )
+    {
+        if( !isRemote )
+        {
+            pLocalCandidate->iceCandidateType = ICE_CANDIDATE_TYPE_PEER_REFLEXIVE;
+            memcpy( &pLocalCandidate->ipAddress,
+                    pIpAddr,
+                    sizeof( IceIPAddress_t ) );
+        }
+        else
+        {
+            foundDuplicatedCandidate = Ice_FindCandidateFromIp( pIceAgent,
+                                                                *pIpAddr,
+                                                                isRemote,
+                                                                &pPeerReflexiveCandidate );
+
+            if( !foundDuplicatedCandidate )
+            {
+                retStatus = Ice_AddRemoteCandidate( pIceAgent,
+                                                    ICE_CANDIDATE_TYPE_PEER_REFLEXIVE,
+                                                    &pPeerReflexiveCandidate,
+                                                    *pIpAddr,
+                                                    0,
+                                                    priority );
+            }
+        }
+    }
+
+    return retStatus;
+}
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_ComputeCandidatePriority - Compute the candidate priority */
+
+static uint32_t Ice_ComputeCandidatePriority( IceCandidate_t * pIceCandidate )
+{
+    uint32_t typePreference = 0, localPreference = 0;
+
+    if( pIceCandidate != NULL )
+    {
+        switch( pIceCandidate->iceCandidateType )
+        {
+            case ICE_CANDIDATE_TYPE_HOST:
+                typePreference = ICE_PRIORITY_HOST_CANDIDATE_TYPE_PREFERENCE;
+                break;
+
+            case ICE_CANDIDATE_TYPE_SERVER_REFLEXIVE:
+                typePreference = ICE_PRIORITY_SERVER_REFLEXIVE_CANDIDATE_TYPE_PREFERENCE;
+                break;
+
+            case ICE_CANDIDATE_TYPE_PEER_REFLEXIVE:
+                typePreference = ICE_PRIORITY_PEER_REFLEXIVE_CANDIDATE_TYPE_PREFERENCE;
+                break;
+
+            case ICE_CANDIDATE_TYPE_RELAYED:
+                typePreference = ICE_PRIORITY_RELAYED_CANDIDATE_TYPE_PREFERENCE;
+                break;
```

Comment:
default case missing

Reply:
They do not have the default case in the WebRTC SDK as well, hence followed the same method. Now the default case has been added.